### PR TITLE
Fix delete cycle progress key and race condition

### PR DESCRIPTION
**Pull request by nllong**
When calling the `delete` mixin that requires a progress key (e.g., delete_cycle), the progress key wasn't being tracked. This PR updates the tracking of such cases.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ repos:
       - id: mixed-line-ending
         args: ["--fix=auto"]
   - repo: https://github.com/pre-commit/mirrors-autopep8
-    rev: v2.0.1
+    rev: v2.0.2
```
Comment (nllong) [inline .pre-commit-config.yaml:25]: `pre-commit autoupdate`

File: pyseed/seed_client.py [hunk 2]
```diff
@@ -652,7 +652,13 @@ def delete_cycle(self, cycle_id: str) -> dict:
         Returns:
             dict:
         """
-        return self.client.delete(cycle_id, endpoint="cycles")
+        result = self.client.delete(cycle_id, endpoint="cycles")
+        progress_key = result.get("progress_key", None)
```
Comment (nllong) [inline pyseed/seed_client.py:656]: need to track the delete progress key

File: pyseed/seed_client_base.py [hunk 3]
```diff
@@ -220,9 +220,16 @@ def _check_response(self, response, *args, **kwargs):
         # This is superfluous as status codes should be used to indicate an
         # error, but they are not always set correctly.
         elif isinstance(response.json(), dict):
+
             status_field = response.json().get('status', None)
+            has_progress_key = 'progress_key' in response.json().keys()
             if status_field:
-                if status_field == 'error':
+                if has_progress_key:
```
Comment (nllong) [inline pyseed/seed_client_base.py:227]: very silly edge case--SEED isn't consistent with all its API calls/returns.

File: pyseed/seed_client_base.py [hunk 4]
```diff
@@ -504,9 +511,10 @@ def delete(self, pk, endpoint=None, data_name=None, **kwargs):
         url = add_pk(self.urls[endpoint], pk, required=kwargs.pop('required_pk', True), slash=True)
         url = _replace_url_args(url, url_args)
         response = super(DeleteMixin, self)._delete(url=url, **kwargs)
-        # delete should return 204 and no content
+        # delete should return 204 and no content, unless it is a background task
         if response.status_code != requests.codes.no_content:
             self._check_response(response, **kwargs)
+            return self._get_result(response, data_name=data_name, **kwargs)
```
Comment (nllong) [inline pyseed/seed_client_base.py:517]: in some cases there are results with `delete` calls

**Review by axelstudios — APPROVED** [review]

**Pull request merged**